pkg/exprparser/interpreter.go

```diff
@@ -170,6 +170,9 @@ func (impl *interperterImpl) evaluateIndexAccess(indexAccessNode *actionlint.Ind
 	case reflect.Int:
 		switch leftValue.Kind() {
 		case reflect.Slice:
+			if rightValue.Int() >= int64(leftValue.Len()) {
+				return nil, nil
+			}
```

Comment:
Please add a test case for this

Reply:
Sure, sounds like a good idea. Whatever you can do to help us get closer to GHA in terms of compatibility would be awesome 🚀 
But let's do that then in a separate PR.